### PR TITLE
Gcs multifile related test cases

### PR DESCRIPTION
**Pull request by sumitsri12**
@sumitnigamcs

### CODE REVIEW

File: src/e2e-test/features/GCSMultifileError.feature [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  @TC-Error-2
+  Scenario: Verify Error validation in path field
+    Given Open Datafusion Project to configure pipeline
+    When Source is GCS bucket
+    When Target is BigQuery
+    Then Link Source and Sink to establish connection
+    Then Enter the BigQuery Properties for table "Employeedata" amd dataset "test_automation" for source
+    Then Close the BigQuery Properties
+    Then Enter the Gcs Multifile Properties for table "gs://cdf-athena" and format "avro"
+    Then Verify path validation
```
Comment (sumitnigamcs) [inline src/e2e-test/features/GCSMultifileError.feature:22]: looks like checkstyle issues are there, please verify

File: src/e2e-test/features/GcsMultifile.feature [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    Then Close Gcs Multifile Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open Logs
+    Then Verify the pipeline status is "Succeeded"
+    Then validate successMessage is displayed
+
+
+
```
Comment (sumitnigamcs) [inline src/e2e-test/features/GcsMultifile.feature:21]: remove the extra lines here

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/package-info.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Package contains the stepDesign for the GCP features.
+ */
+package io.cdap.plugin.gcsmultifile.actions;
+
+
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/package-info.java:6]: remove extra line here

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package io.cdap.plugin.gcsmultifile.locators;
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java:1]: looks like there are no licenses in the files. checkstyle not run it seems

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  public static WebElement closeButton;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='property-row-error' and contains(text(),'referenceName')]")
+  public static WebElement referenceError;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='property-row-error' and contains(text(),'path')]")
+  public static WebElement pathError;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='project' and @class='MuiInputBase-input']")
+  public static WebElement ProjectId;
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java]: camelcasing correction

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/package-info.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Package contains the stepDesign for the GCP features.
+ */
+package io.cdap.plugin.gcsmultifile.locators;
+
+
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/package-info.java:6]: remove extra line here

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  @FindBy(how = How.XPATH, using = "//*[text()='Source ']")
+  public static WebElement source;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='valium-banner-hydrator']")
+  public static WebElement pipelineSaveSuccessBanner;
+
+}
+
+
+
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java:62]: remove extra lines here

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+  public static String getPathError() {
+    return GCSMultifileLocators.pathError.getText();
+  }
+
+  public static String getPathErrorColor() {
+    return GCSMultifileLocators.pathError.getCssValue(COLOR);
+  }
+
+  public static void enterdatasetProjectId() throws IOException {
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java:77]: camelcasing correction

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException, InterruptedException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = new PrintWriter(BeforeActions.myObj);
+    out.println(CdfPipelineRunAction.captureRawLogs());
+    out.close();
```
Comment (sumitnigamcs) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java:155]: close this in the finally block

File: src/e2e-test/features/GCSMultifileError.feature (hunk 1 above)
Comment (sawantpritam) [inline src/e2e-test/features/GCSMultifileError.feature:22]: No New line at end of file. Add empty line

File: src/e2e-test/features/GcsMultifile.feature [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+  @TC-GCS-GCSMultiflow
+  Scenario: User is able to Login and confirm data is getting transferred from BigQuery to GcsMultifile
```
Comment (sawantpritam) [inline src/e2e-test/features/GcsMultifile.feature:23]: Change scenario description. Steps are for GCS to GCSMultifile

File: src/e2e-test/features/GcsMultifile.feature [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+    Then Close Gcs Multifile Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open Logs
+    Then Verify the pipeline status is "Succeeded"
+    Then validate successMessage is displayed
+
+  @TC-MultipleSource-GCSMultiflow
+  Scenario: User is able to Login and confirm data is getting transferred from BigQuery to GcsMultifile
```
Comment (sawantpritam) [inline src/e2e-test/features/GcsMultifile.feature:40]: Change scenario description.

File: src/e2e-test/features/GcsMultifile.feature [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+    Then Close Gcs Multifile Properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open Logs
+    Then Verify the pipeline status is "Succeeded"
+    Then validate successMessage is displayed
+
+  @TC-FileFormat-tsv
+  Scenario: User is able to Login and confirm data is getting transferred from BigQuery to GcsMultifile
```
Comment (sawantpritam) [inline src/e2e-test/features/GcsMultifile.feature:62]: Change scenario description... check for all subsequent scenarios as well.

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java:77]: Check camelcase. enterdatasetProjectId -> enterDatasetProjectId

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java [hunk 14]
```diff
@@ -0,0 +1,85 @@
+package io.cdap.plugin.gcsmultifile.actions;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/actions/BigQueryMulifileActions.java:1]: Add licence

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
+package io.cdap.plugin.gcsmultifile.locators;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/locators/GCSMultifileLocators.java:1]: Add Licence.

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java [hunk 16]
```diff
@@ -0,0 +1,212 @@
+package io.cdap.plugin.gcsmultifile.stepsdesign;
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java:1]: Add licence.

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+  public void closeGcsMultifileProperties() {
+    BigQueryMulifileActions.closeGcsMultifile();
+  }
+
+  @Then("verify the schema in output")
+  public void verifyTheSchemaInOutput() {
+    CdfGcsActions.validateSchema();
+  }
+
+  @Then("Verify the pipeline status in each case")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java:108]: remove if unused.

File: src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException, InterruptedException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = new PrintWriter(BeforeActions.myObj);
+    out.println(CdfPipelineRunAction.captureRawLogs());
+    out.close();
+  }
+
+  @Then("validate successMessage is displayed")
```
Comment (sawantpritam) [inline src/e2e-test/java/io/cdap/plugin/gcsmultifile/stepsdesign/GCSMultifileConnector.java:158]: validate ->Validate

File: src/e2e-test/resources/errorMessage.properties [hunk 19]
```diff
@@ -0,0 +1,5 @@
+errorMessageTemporaryBucket=Bucket name can only contain lowercase letters, numbers, '.', '_', and '-'.
+errorMessageDataset=Required property 'dataset' has no value.
+errorMessageReference=Required property 'referenceName' has no value.
+errorMessageColor=#a40403
+errorMessagePath=Required property 'path' has no value.
```
Comment (sawantpritam) [inline src/e2e-test/resources/errorMessage.properties:5]: No new line at the end of file. add empty line.

**Review by sawantpritam — CHANGES_REQUESTED** [review]

**Review by sumitnigamcs — CHANGES_REQUESTED** [review]

File: src/e2e-test/features/GcsMultifile.feature (hunk 2 above)
Comment (ksahilCF) [reply src/e2e-test/features/GcsMultifile.feature:21]: all the scenario have same description

**Pull request closed without merging**